### PR TITLE
PHP 8.0 support

### LINKED ISSUE #66: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by snapshotpl**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Drops php lower than php 7.3 and add support for php 8.

Closes #66

### CODE REVIEW

File: test/ApplicationTest.php [hunk 1]
```diff
@@ -501,7 +507,7 @@ public function testOnDispatchErrorEventPassedToTriggersShouldBeTheOriginalOne()
     }
 
     /**
-     * @group 2981
+     * @group #2981
```
Comment (snapshotpl) [inline test/ApplicationTest.php]: Related to https://github.com/sebastianbergmann/phpunit/issues/4489

**Comment by snapshotpl** [comment]
Build on PHP 8 fails because https://github.com/laminas/laminas-stratigility/blob/2.2.2/src/MiddlewarePipe.php#L261

Shall we drop stratigilty v2? Or...

File: test/ApplicationTest.php (hunk 1 above)
Comment (Xerkus) [reply test/ApplicationTest.php]: Those should be removed. They predate repository split of 2.5 and are irrelevant

**Comment by snapshotpl** [comment]
I see now this PR https://github.com/laminas/laminas-mvc/pull/51
So imo there are required steps to make this PR works:
* merge and release 3.2 https://github.com/laminas/laminas-mvc/pull/51
* Merge and release 2.0 https://github.com/laminas/laminas-mvc-middleware/pull/3
* Resolve somehow this PR to allow use laminas-mvc-middleware in v1 and 2 to keep BC

**Comment by Xerkus** [comment]
Stratigility 2 is not installable on PHP 8 normally and it is an optional dependency. I updated tests for it to skip on PHP 8.

**Review by Xerkus — APPROVED** [review]

**Pull request merged**